evap/results/views.py

```diff
@@ -94,9 +94,9 @@ def update_template_cache_of_published_evaluations_in_course(course):
 
 def get_evaluations_with_prefetched_data(evaluations):
     if isinstance(evaluations, QuerySet):
-        participant_counts = evaluations.annotate(num_participants=Count("participants")).values_list("num_participants", flat=True)
-        voter_counts = evaluations.annotate(num_voters=Count("voters")).values_list("num_voters", flat=True)
-        course_evaluations_counts = evaluations.annotate(num_course_evaluations=Count("course__evaluations")).values_list("num_course_evaluations", flat=True)
+        participant_counts = evaluations.annotate(num_participants=Count("participants")).order_by('pk').values_list("num_participants", flat=True)
+        voter_counts = evaluations.annotate(num_voters=Count("voters")).order_by('pk').values_list("num_voters", flat=True)
+        course_evaluations_counts = evaluations.annotate(num_course_evaluations=Count("course__evaluations")).order_by('pk').values_list("num_course_evaluations", flat=True)
```

Comment:
I wonder whether these annotations could be grouped into one. When I tried it locally, I discovered that [one must use `Count(…, distinct=True)` when annotating with multiple counts](https://docs.djangoproject.com/en/3.0/topics/db/aggregation/#combining-multiple-aggregations).
I could also imagine that the the later `for in zip()` loop could be expressed with a single annotation with a `Coalesce` for participants and voters.
But these are just wild thoughts with unprofiled guesses :D

Reply:
it seemed to work! unfortunately, it was about 2x slower than doing individual queries...

Reply:
Ah, bummer. Properly it is because Django uses joins instead of subqueries which the database might not be able to optimize correctly…